File: library/src/main/java/io/ashdavies/rx/tasks/CompletableTaskListenerFactory.java
package io.ashdavies.rx.tasks;

import com.google.android.gms.tasks.OnFailureListener;
import com.google.android.gms.tasks.OnSuccessListener;
import io.reactivex.CompletableEmitter;

class CompletableTaskListenerFactory implements TaskListenerFactory<Void, CompletableEmitter> {

  @Override
  public OnSuccessListener<Void> createOnSuccessListener(CompletableEmitter emitter) {
    return new CompletableEmitterSuccessListener(emitter);
  }

  @Override
  public OnFailureListener createOnFailureListener(CompletableEmitter emitter) {
    return new CompletableEmitterFailureListener(emitter);
  }
}

File: src/main/kotlin/io/ashdavies/rx/rxtasks/CompletableTaskListenerFactory.kt
package io.ashdavies.rx.rxtasks

import io.reactivex.CompletableEmitter

internal class CompletableTaskListenerFactory : TaskListenerFactory<Void, CompletableEmitter> {

  override fun createOnSuccessListener(emitter: CompletableEmitter) = CompletableEmitterSuccessListener(emitter)

  override fun createOnFailureListener(emitter: CompletableEmitter)= CompletableEmitterFailureListener(emitter)
}
